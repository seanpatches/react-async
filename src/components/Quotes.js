import React from 'react';
import PropTypes from 'prop-types';
import Quote from './Quote.js';

function Quotes({ quotes }) {
  const quoteList = quotes.map(quote => (
    <li key={quote.quote}>
      <Quote quote={quote} />
    </li>
  ));


  return ( 
    <ul>
      {quoteList}
    </ul>
  );
}

Quotes.propTypes = {
  quotes: PropTypes.array.isRequired
};

export default Quotes;
