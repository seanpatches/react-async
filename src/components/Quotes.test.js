import React from 'react';
import { shallow } from 'enzyme';
import Quotes from './Quotes';

describe('Quotes component', () => {
  it('renders Quotes', () => {
    const quotes = [{
      character: 'Fry',
      image: '/image',
      quote: 'What??'
    },
    {
      character: 'Me',
      image: '/image2',
      quote: 'Why??'
    }];

    const wrapper = shallow(<Quotes quotes={quotes} />);
    expect(wrapper).toMatchSnapshot();
  });
});
