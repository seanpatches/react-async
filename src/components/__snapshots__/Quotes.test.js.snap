// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Quotes component renders Quotes 1`] = `
<ul>
  <li
    key="Fry"
  >
    <Quote
      quote={
        Object {
          "character": "Fry",
          "image": "/image",
          "quote": "What??",
        }
      }
    />
  </li>
  <li
    key="Me"
  >
    <Quote
      quote={
        Object {
          "character": "Me",
          "image": "/image2",
          "quote": "Why??",
        }
      }
    />
  </li>
</ul>
`;
